### ISSUE TITLE
MPU9250 I AM 0x0 I should be 0x71. Could not connect to MPU9250 (Pro Mini 3v)

### ISSUE THREAD

**Issue by antpgomes** [kriswiner/MPU9250#211]
Hi Everyone

I read through all the posts and for the life of me I can't figure this out.

I have a sparkfun mpu9250, i see it as i2c address 0x68 on the scanner but i can't seem to get the example code to work.

I changed the following line:
byte c = myIMU.readByte(0x68, WHO_AM_I_MPU9250); to reflect the address but still no luck.

Any help would be appreciated.

EDIT:
I also tried pulling the CS pin high to no avail

**Comment by lukecaj** [comment]
I'm having the same sort of issue on my Arduino Zero, except I get a value of 0x1.  I see 0x68 returned in the I2C scan list, but not 0x71.

**Comment by kriswiner** [comment]
0x68 is the MPU9250 I2C addresYou have to read the WHO_AM_I register of the
MPU9250 to get 0x71.

**Comment by yNosGR** [comment]
I am having the same problem with an amazon MPU9250 board on an 8266.
@kriswiner, the line @antpgomes references above:
byte c = myIMU.readByte(0x68, WHO_AM_I_MPU9250);
reads i2c address 0x68 and the WHO_AM_I_MPU9250 register (0x75) into byte c.

**Comment by kriswiner** [comment]
Are there pullups on the board? Are they working? If you scan for I2C
devices what do you see?

**Comment by yNosGR** [comment]
When I scan for devices, I see the bmp180 and the MPU9250; I can poll the bmp180 w/o issue - maybe the 9250 wants to be the only one on the bus?
-------8<--snip-->8-------
Scanning...
I2C device found at address 0x68  !
I2C device found at address 0x77  !
done
-------8<--snip-->8-------

**Comment by kriswiner** [comment]
No, multiple devices can be on the bus, that is the whole point of I2C.

And when you read the WHO_AM_I register, what do you get?

**Comment by yNosGR** [comment]
It replies with a 0x0, like OP.

**Comment by kriswiner** [comment]
Maybe a crappy sensor, or bad I2C API

**Comment by yNosGR** [comment]
Since I am using the arduino IDE, I'll blame it on the sensor.

**Comment by kriswiner** [comment]
Can you try another sensor then. it could be something wrong with your
Wire.h calls.

**Comment by yNosGR** [comment]
I just got a sparkfun board in, wired it up with the same results. The puzzling part is that I am able to communicate with both the bmp180 and bmp280 sensors have here w/o issue. I am not sure what is different about the mpu9250. Oh.. I forgot about the DS1307 on there as well..

**Comment by kriswiner** [comment]
Do you have pullups enabled on the Sparkfun board?

**Comment by yNosGR** [comment]
Would I need them on all devices? Currently they are on the bmp180 board, the last device in the chain.

**Comment by kriswiner** [comment]
No, just somewhere on the I2C bus. Do you have to enable the I2C bus some
how on the Sparkfun board? Did you follow their tutorial? Maybe a bad
board? Better ask Sparkfun...
